toolkit/offchain/src/untyped_plutus.rs

```diff
@@ -10,31 +9,56 @@ use uplc::plutus_data;
 /// Currently there is no other known option to apply parameters to plutus script in Rust.
 ///
 /// Parameters:
-/// * `params` - parameters to apply to the script
+/// * `param` - single PlutusData parameter to apply to the script
 /// * `plutus_script_raw` - raw plutus script in CBOR format, like in `RawScripts.purs` in smart-contracts repository
-pub(crate) fn apply_params_to_script<T: ToDatum>(
-	params: &T,
+pub fn apply_params_to_script<T: ToDatum>(
+	param: &T,
 	plutus_script_raw: &[u8],
 ) -> Result<Vec<u8>, anyhow::Error> {
-	let params: uplc::PlutusData =
-		plutus_data(&minicbor::to_vec(params.to_datum()).expect("to_vec is Infallible"))
-			.expect("trasformation from PC Datum to pallas PlutusData can't fail");
+	let param = datum_to_uplc_plutus_data(param);
+	apply_uplc_params_to_script(&[&param], plutus_script_raw)
+}
 
+/// Parameters:
+/// * `params` - list of PlutusData parameters to apply to the script, not necessarily the "partner-chain params"
+/// * `plutus_script_raw` - raw plutus script in CBOR format, like in `RawScripts.purs` in smart-contracts repository
+pub fn apply_uplc_params_to_script(
```

Comment:
```suggestion
pub fn apply_params_to_script(
```

Reply:
Yeah, I'll go for it. `uplc` was used to make distinction from `uplc::PlutusData` and our `ToDatum` At some point we should settle with only one implementation.